pkg/agent/controller/bgp/controller.go

```diff
@@ -571,56 +588,95 @@ func (c *Controller) addServiceRoutes(advertisement *v1alpha1.ServiceAdvertiseme
 		if ipTypes.Has(v1alpha1.ServiceIPTypeClusterIP) {
 			if internalLocal && hasLocalEndpoints || !internalLocal {
 				for _, clusterIP := range svc.Spec.ClusterIPs {
-					serviceIPs = append(serviceIPs, clusterIP)
+					if c.enabledIPv4 && utilnet.IsIPv4String(clusterIP) {
+						allRoutes[bgp.Route{Prefix: clusterIP + ipv4Suffix}] = RouteMetadata{
+							Type:      ServiceClusterIP,
+							K8sObjRef: svc.GetNamespace() + "/" + svc.GetName(),
```

Comment:
nit: compute the "k8sObjRef" once at the beginning of the `for _, svc := range services` loop